lib/lightning/workflows.ex

```diff
@@ -190,4 +190,118 @@ defmodule Lightning.Workflows do
       Repo.update_all(workflow_jobs_query, set: [enabled: false])
     end)
   end
+
+  defp filter_digest(digest) do
+    case digest do
+      :daily ->
+        dynamic([r], fragment("age(now(), ?) <= '1 days'", r.finished_at))
+
+      :weekly ->
+        dynamic([r], fragment("age(now(), ?) <= '7 days'", r.finished_at))
```

Comment:
I think we should avoid using the word digest here, we have an opportunity to have a more generic function like:

`finished_before_query` or similar.

```
def finished_before_query(period) do
  period = [case period do
    :daily -> 1
    ...
  end |> to_string(), "days"]
  |> Enum.join(" ")

  dynamic([r], fragment("age(now(), ?) <= ?", r.finished_at, period))
end
```

Note that I'm passing in the period as well, this will allow Postgres to get more of an opportunity to reuse a query plan (it will cache one query instead of 3).

Also it looks like this query returns _all_ runs before a given date. Do we not need it to be between two dates?  
And I see you're using `age(now() ..` that makes me a little nervous because:

1. After the TZ issues we saw recently with Mtuchi's setup
2. Postgres will need to cast all of the `finished_at` values into ages in order to compare them - removing the chance of any index being used.
  I think we should create the 'correct' date using Timex or similar and pass that in as a `where: r.finished_at >= ^from_date`.